overlord/devicestate/handlers_install.go

```diff
@@ -989,11 +1032,17 @@ func (m *DeviceManager) doFactoryResetRunSystem(t *state.Task, _ *tomb.Tomb) err
 		UnpackedGadgetDir: gadgetDir,
 	}
 	timings.Run(perfTimings, "boot-make-runnable", "Make target system runnable", func(timings.Measurer) {
-		err = bootMakeRunnable(deviceCtx.Model(), bootWith, nil)
+		err = bootMakeRunnableAfterDataReset(deviceCtx.Model(), bootWith, trustedInstallObserver)
 	})
 	if err != nil {
 		return fmt.Errorf("cannot make system runnable: %v", err)
 	}
+
+	// leave a marker that factory reset was performed
+	factoryResetMarker := filepath.Join(dirs.SnapDeviceDirUnder(boot.InstallHostWritableDir), "factory-reset")
+	if err := writeFactoryResetMarker(factoryResetMarker, useEncryption); err != nil {
```

Comment:
In theory we need the sealed key file to exist in order to calculate its hash for the marker. A reboot here would mean that  we come up in a system which would not detect it went through a factory reset.

An alternative is to not have the hash of the sealed key, which means that the marker can be written before making the run system bootable. In case of an unexpected reboot we would still be good.